Source/Core/InputCommon/ControllerEmu/StickGate.cpp

```diff
@@ -118,6 +154,7 @@ std::optional<u32> SquareStickGate::GetIdealCalibrationSampleCount() const
 ReshapableInput::ReshapableInput(std::string name_, std::string ui_name_, GroupType type_)
     : ControlGroup(std::move(name_), std::move(ui_name_), type_)
 {
+  AddVirtualNotchSetting(&m_virtual_notch_setting, 45);
```

Comment:
Unfortunately this will give us a "Notches" setting for things that shouldn't have it. (emulated Wii Remote's Swing, Tilt, etc.).
The solution might be to add an `OctagonAnalogStick` class inheriting `AnalogStick` and replace relevant occurrences.

Reply:
I've spent the morning trying to work out the best way to go about doing this but I'm really struggling here.

My idea was to move the `AddVirtualNotchSetting` and  `GetVirtualNotchSize` implementation into the `OctagonalAnalogStick`  implementation. I changed the `GetVirtualNotchSize` to be a virtual function on `StickGate` and tried to add a default implementation on the base `ReshapableInput` class but it seems like the compiler is expecting it to be implemented across the different `ControlGroup` classes. Is there no way to implement a default behaviour that is specifically overrided by the Octagonal stick? Maybe I'm going about this in the wrong way I really just don't know!

Here is a diff of my current changes:
https://diffy.org/diff/wadmi1m9haj6oqawoisk0529

And here is the error output:
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2259	'ControllerEmu::Force': cannot instantiate abstract class	E:\Users\Nick\Documents\Code\dolphin\Build\Release\dolphin	E:\Users\Nick\Documents\Code\dolphin\Source\Core\Core\HW\WiimoteEmu\Extension\Nunchuk.cpp	49	
Error	C2259	'ControllerEmu::Force': cannot instantiate abstract class	E:\Users\Nick\Documents\Code\dolphin\Build\Release\dolphin	E:\Users\Nick\Documents\Code\dolphin\Source\Core\Core\HW\WiimoteEmu\Extension\Nunchuk.cpp	49	
Error	C2259	'ControllerEmu::Tilt': cannot instantiate abstract class	E:\Users\Nick\Documents\Code\dolphin\Build\Release\dolphin	E:\Users\Nick\Documents\Code\dolphin\Source\Core\Core\HW\WiimoteEmu\Extension\Nunchuk.cpp	52	
Error	C2259	'ControllerEmu::Tilt': cannot instantiate abstract class	E:\Users\Nick\Documents\Code\dolphin\Build\Release\dolphin	E:\Users\Nick\Documents\Code\dolphin\Source\Core\Core\HW\WiimoteEmu\Extension\Nunchuk.cpp	52	
Error	C2259	'ControllerEmu::Cursor': cannot instantiate abstract class	E:\Users\Nick\Documents\Code\dolphin\Build\Release\dolphin	E:\Users\Nick\Documents\Code\dolphin\Source\Core\Core\HW\WiimoteEmu\WiimoteEmu.cpp	210	
Error	C2259	'ControllerEmu::Cursor': cannot instantiate abstract class	E:\Users\Nick\Documents\Code\dolphin\Build\Release\dolphin	E:\Users\Nick\Documents\Code\dolphin\Source\Core\Core\HW\WiimoteEmu\WiimoteEmu.cpp	210	
Error	C2259	'ControllerEmu::Force': cannot instantiate abstract class	E:\Users\Nick\Documents\Code\dolphin\Build\Release\dolphin	E:\Users\Nick\Documents\Code\dolphin\Source\Core\Core\HW\WiimoteEmu\WiimoteEmu.cpp	211	
Error	C2259	'ControllerEmu::Force': cannot instantiate abstract class	E:\Users\Nick\Documents\Code\dolphin\Build\Release\dolphin	E:\Users\Nick\Documents\Code\dolphin\Source\Core\Core\HW\WiimoteEmu\WiimoteEmu.cpp	211	
Error	C2259	'ControllerEmu::Tilt': cannot instantiate abstract class	E:\Users\Nick\Documents\Code\dolphin\Build\Release\dolphin	E:\Users\Nick\Documents\Code\dolphin\Source\Core\Core\HW\WiimoteEmu\WiimoteEmu.cpp	212	
Error	C2259	'ControllerEmu::Tilt': cannot instantiate abstract class	E:\Users\Nick\Documents\Code\dolphin\Build\Release\dolphin	E:\Users\Nick\Documents\Code\dolphin\Source\Core\Core\HW\WiimoteEmu\WiimoteEmu.cpp	212	
```

Reply:
I think I got it figured out (b556c798)

Reply:
Oh. I didn't realize we already had an `OctagonAnalogStick` class. :P